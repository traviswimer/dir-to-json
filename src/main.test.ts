import path from "path";

import dirToJson, { INVALID_PATH_ERROR, INVALID_OPTIONS_ERROR } from "./main";

test(`rejects if "path" is not a valid string`, () => {
	// @ts-ignore: This is testing an invalid parameter
	return expect(dirToJson()).rejects.toMatch(INVALID_PATH_ERROR);
});
test(`rejects if "options" is not a valid object`, () => {
	return expect(
		// @ts-ignore: This is testing an invalid parameter
		dirToJson("a/fake/path", "This should be an object")
	).rejects.toMatch(INVALID_OPTIONS_ERROR);
});

describe("single file", () => {
	let filePath: string;
	beforeEach(() => {
		filePath = path.resolve(__dirname, "../fixtures/file_1.txt");
	});
	test("should match snapshot", async function () {
		const data: any = await dirToJson(filePath);
		expect(data).toMatchSnapshot();
	});

	test("should resolve with no children", async function () {
		const data: any = await dirToJson(filePath);
		expect(data.children).toBeUndefined;
	});
});

describe("directory with one file", () => {
	let filePath: string;
	beforeEach(() => {
		filePath = path.resolve(__dirname, "../fixtures/dir-1");
	});
	test("should match snapshot", async function () {
		const data: any = await dirToJson(filePath);
		expect(data).toMatchSnapshot();
	});

	test("should resolve with one child", async function () {
		const data: any = await dirToJson(filePath);
		expect(data.children.length).toEqual(1);
	});
});

describe("directory with multiple files", () => {
	let filePath: string;
	beforeEach(() => {
		filePath = path.resolve(__dirname, "../fixtures/dir-2");
	});
	test("should match snapshot", async function () {
		const data: any = await dirToJson(filePath);
		expect(data).toMatchSnapshot();
	});

	test("should resolve with three children", async function () {
		const data: any = await dirToJson(filePath);
		expect(data.children.length).toEqual(3);
	});
});

describe("directory with single sub directory", () => {
	let filePath: string;
	beforeEach(() => {
		filePath = path.resolve(__dirname, "../fixtures/dir-3");
	});
	test("should match snapshot", async function () {
		const data: any = await dirToJson(filePath);
		expect(data).toMatchSnapshot();
	});

	test("should resolve with a directory child", async function () {
		const data: any = await dirToJson(filePath);
		expect(data.children[0].children.length).toEqual(1);
	});
});

describe("directory with multiple levels", () => {
	let filePath: string;
	beforeEach(() => {
		filePath = path.resolve(__dirname, "../fixtures/dir-4");
	});
	test("should match snapshot", async function () {
		const data: any = await dirToJson(filePath);
		expect(data).toMatchSnapshot();
	});

	test("should resolve with a directory child", async function () {
		const data: any = await dirToJson(filePath);
		expect(
			data.children[0].children[0].children[0].children[0].children.length
		).toEqual(2);
	});
});

describe("options.sortType", () => {
	let filePath: string;
	beforeEach(() => {
		filePath = path.resolve(__dirname, "../fixtures/dir-5");
	});
	test("should match snapshot when true", async function () {
		const data: any = await dirToJson(filePath, { sortType: true });
		expect(data).toMatchSnapshot();
	});
	test("should match snapshot when false", async function () {
		const data: any = await dirToJson(filePath, { sortType: false });
		expect(data).toMatchSnapshot();
	});
});
