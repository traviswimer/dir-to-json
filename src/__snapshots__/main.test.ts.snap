// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`directory with multiple files should match snapshot 1`] = `
Object {
  "children": Array [
    Object {
      "name": "file-2-1.txt",
      "parent": "",
      "path": "file-2-1.txt",
      "type": "file",
    },
    Object {
      "name": "file-2-2.txt",
      "parent": "",
      "path": "file-2-2.txt",
      "type": "file",
    },
    Object {
      "name": "file-2-3.txt",
      "parent": "",
      "path": "file-2-3.txt",
      "type": "file",
    },
  ],
  "name": "dir-2",
  "parent": "..",
  "path": "",
  "type": "directory",
}
`;

exports[`directory with multiple levels should match snapshot 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [
                    Object {
                      "name": "file-4-1-1-1-1_1.txt",
                      "parent": "dir-4-1/dir-4-1-1/dir-4-1-1-1 /dir-4-1-1-1-1",
                      "path": "dir-4-1/dir-4-1-1/dir-4-1-1-1 /dir-4-1-1-1-1/file-4-1-1-1-1_1.txt",
                      "type": "file",
                    },
                    Object {
                      "name": "file-4-1-1-1-1_2.txt",
                      "parent": "dir-4-1/dir-4-1-1/dir-4-1-1-1 /dir-4-1-1-1-1",
                      "path": "dir-4-1/dir-4-1-1/dir-4-1-1-1 /dir-4-1-1-1-1/file-4-1-1-1-1_2.txt",
                      "type": "file",
                    },
                  ],
                  "name": "dir-4-1-1-1-1",
                  "parent": "dir-4-1/dir-4-1-1/dir-4-1-1-1 ",
                  "path": "dir-4-1/dir-4-1-1/dir-4-1-1-1 /dir-4-1-1-1-1",
                  "type": "directory",
                },
                Object {
                  "children": Array [
                    Object {
                      "name": "file-4-1-1-1-2_1.txt",
                      "parent": "dir-4-1/dir-4-1-1/dir-4-1-1-1 /dir-4-1-1-1-2",
                      "path": "dir-4-1/dir-4-1-1/dir-4-1-1-1 /dir-4-1-1-1-2/file-4-1-1-1-2_1.txt",
                      "type": "file",
                    },
                    Object {
                      "name": "file-4-1-1-1-2_2.txt",
                      "parent": "dir-4-1/dir-4-1-1/dir-4-1-1-1 /dir-4-1-1-1-2",
                      "path": "dir-4-1/dir-4-1-1/dir-4-1-1-1 /dir-4-1-1-1-2/file-4-1-1-1-2_2.txt",
                      "type": "file",
                    },
                  ],
                  "name": "dir-4-1-1-1-2",
                  "parent": "dir-4-1/dir-4-1-1/dir-4-1-1-1 ",
                  "path": "dir-4-1/dir-4-1-1/dir-4-1-1-1 /dir-4-1-1-1-2",
                  "type": "directory",
                },
                Object {
                  "name": "file-4-1-1-1-1_1.txt",
                  "parent": "dir-4-1/dir-4-1-1/dir-4-1-1-1 ",
                  "path": "dir-4-1/dir-4-1-1/dir-4-1-1-1 /file-4-1-1-1-1_1.txt",
                  "type": "file",
                },
              ],
              "name": "dir-4-1-1-1 ",
              "parent": "dir-4-1/dir-4-1-1",
              "path": "dir-4-1/dir-4-1-1/dir-4-1-1-1 ",
              "type": "directory",
            },
            Object {
              "name": "file-4-1-1_1.txt",
              "parent": "dir-4-1/dir-4-1-1",
              "path": "dir-4-1/dir-4-1-1/file-4-1-1_1.txt",
              "type": "file",
            },
          ],
          "name": "dir-4-1-1",
          "parent": "dir-4-1",
          "path": "dir-4-1/dir-4-1-1",
          "type": "directory",
        },
        Object {
          "name": "file-4-1_1.txt",
          "parent": "dir-4-1",
          "path": "dir-4-1/file-4-1_1.txt",
          "type": "file",
        },
      ],
      "name": "dir-4-1",
      "parent": "",
      "path": "dir-4-1",
      "type": "directory",
    },
    Object {
      "name": "file-4_1.txt",
      "parent": "",
      "path": "file-4_1.txt",
      "type": "file",
    },
  ],
  "name": "dir-4",
  "parent": "..",
  "path": "",
  "type": "directory",
}
`;

exports[`directory with one file should match snapshot 1`] = `
Object {
  "children": Array [
    Object {
      "name": "file-1-1.txt",
      "parent": "",
      "path": "file-1-1.txt",
      "type": "file",
    },
  ],
  "name": "dir-1",
  "parent": "..",
  "path": "",
  "type": "directory",
}
`;

exports[`directory with single sub directory should match snapshot 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "name": "file-3-1_1.txt",
          "parent": "dir-3-1",
          "path": "dir-3-1/file-3-1_1.txt",
          "type": "file",
        },
      ],
      "name": "dir-3-1",
      "parent": "",
      "path": "dir-3-1",
      "type": "directory",
    },
    Object {
      "name": "file-3_3.txt",
      "parent": "",
      "path": "file-3_3.txt",
      "type": "file",
    },
  ],
  "name": "dir-3",
  "parent": "..",
  "path": "",
  "type": "directory",
}
`;

exports[`options.sortType should match snapshot when false 1`] = `
Object {
  "children": Array [
    Object {
      "name": "1-file.txt",
      "parent": "",
      "path": "1-file.txt",
      "type": "file",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "name": "1-file.txt",
              "parent": "2-dir/1-dir",
              "path": "2-dir/1-dir/1-file.txt",
              "type": "file",
            },
            Object {
              "name": "2-file.txt",
              "parent": "2-dir/1-dir",
              "path": "2-dir/1-dir/2-file.txt",
              "type": "file",
            },
          ],
          "name": "1-dir",
          "parent": "2-dir",
          "path": "2-dir/1-dir",
          "type": "directory",
        },
        Object {
          "name": "2-file.txt",
          "parent": "2-dir",
          "path": "2-dir/2-file.txt",
          "type": "file",
        },
        Object {
          "children": Array [
            Object {
              "name": "1-file.txt",
              "parent": "2-dir/3-dir",
              "path": "2-dir/3-dir/1-file.txt",
              "type": "file",
            },
            Object {
              "name": "2-file.txt",
              "parent": "2-dir/3-dir",
              "path": "2-dir/3-dir/2-file.txt",
              "type": "file",
            },
          ],
          "name": "3-dir",
          "parent": "2-dir",
          "path": "2-dir/3-dir",
          "type": "directory",
        },
      ],
      "name": "2-dir",
      "parent": "",
      "path": "2-dir",
      "type": "directory",
    },
    Object {
      "name": "3-file.txt",
      "parent": "",
      "path": "3-file.txt",
      "type": "file",
    },
  ],
  "name": "dir-5",
  "parent": "..",
  "path": "",
  "type": "directory",
}
`;

exports[`options.sortType should match snapshot when true 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "name": "1-file.txt",
              "parent": "2-dir/1-dir",
              "path": "2-dir/1-dir/1-file.txt",
              "type": "file",
            },
            Object {
              "name": "2-file.txt",
              "parent": "2-dir/1-dir",
              "path": "2-dir/1-dir/2-file.txt",
              "type": "file",
            },
          ],
          "name": "1-dir",
          "parent": "2-dir",
          "path": "2-dir/1-dir",
          "type": "directory",
        },
        Object {
          "children": Array [
            Object {
              "name": "1-file.txt",
              "parent": "2-dir/3-dir",
              "path": "2-dir/3-dir/1-file.txt",
              "type": "file",
            },
            Object {
              "name": "2-file.txt",
              "parent": "2-dir/3-dir",
              "path": "2-dir/3-dir/2-file.txt",
              "type": "file",
            },
          ],
          "name": "3-dir",
          "parent": "2-dir",
          "path": "2-dir/3-dir",
          "type": "directory",
        },
        Object {
          "name": "2-file.txt",
          "parent": "2-dir",
          "path": "2-dir/2-file.txt",
          "type": "file",
        },
      ],
      "name": "2-dir",
      "parent": "",
      "path": "2-dir",
      "type": "directory",
    },
    Object {
      "name": "1-file.txt",
      "parent": "",
      "path": "1-file.txt",
      "type": "file",
    },
    Object {
      "name": "3-file.txt",
      "parent": "",
      "path": "3-file.txt",
      "type": "file",
    },
  ],
  "name": "dir-5",
  "parent": "..",
  "path": "",
  "type": "directory",
}
`;

exports[`single file should match snapshot 1`] = `
Object {
  "name": "file_1.txt",
  "parent": "..",
  "path": "",
  "type": "file",
}
`;
